realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -456,30 +456,62 @@ public E get(int location) {
      * {@inheritDoc}
      */
     public E first() {
+        return firstImpl(true, null);
+    }
+
+    /**
+    * {@inheritDoc}
+    */
+    public E first(E defaultValue) {
+        return firstImpl(false, defaultValue);
+    }
+
+    private E firstImpl(boolean shouldThrow, E defaultValue) {
```

Comment:
I would simplify this by just using `defaultValue != null` as a substitute for `shouldThrow`. No reason to create a boolean for that.